src/monotonic.c

```diff
@@ -0,0 +1,107 @@
+#include "monotonic.h"
+#include <stddef.h>
+#include <stdlib.h>
+
+/* A standard assert can be compiled out.  This is too low-level for serverAssert().  */
+#define localAssert(_e) do { if (!(_e)) {*((char*)-1) = 'x'; exit(1);} } while(0)
```

Comment:
Thanks Oran reviewing.  Regarding this assertion, I did this because I thought it was the least evil option.
* The standard system `assert` is disabled if `NDEBUG` is defined (which is often true in production systems.
* `redisassert.h` IMO is an **awful** option.  1) it is named the same as the standard POSIX assert, but isn't.  So it can be a source of confusion.  2) `assert` needs to be an extremely low-level function, including nothing.  However `redisassert.h` is dependent on `_serverAssert` and is therefore dependent on `debug.c` which is dependent on stuff like `crc64`????

Regarding the use of `exit(1)`, I simply copied the strategy used in `redisassert.h` and `server.h`.  Note that the `exit` is dead code and is never hit.  A SEGV is created before the `exit`.  The purpose of the `exit` is only to help static analysis tools understand that execution will not continue past that point.

After reading your comment and looking at the docs more, I think a better solution would be to use the standard POSIX `assert(3)` - BUT explicitly `#undef NDEBUG` in this `.c` file.  In that way the program will terminate with standard minimal debug information.  I will change to this approach unless there is violent disagreement.

I am not in favor of returning an error and leaving responsibility to the caller.  This would just pass problems along.  In essence, the assertions in this file are testing for build errors - if these assertions fail, the program can't continue.